### DISCUSSION TITLE
`mypy` telling `Value of type "PdfObject" is not indexable`

### DISCUSSION THREAD

**Discussion by rassie** [py-pdf/pypdf#2140] [Q&A] [answered]
I'm moving from PyPDF2 1.x to current `pypdf` and also trying to use typings in the process. At one point, I need to look into `/OCProperties` dictionary and I try the following (assuming I have a `PdfReader` instance from earlier):

```
root = reader.trailer[TrailerKeys.ROOT]
properties = root[CatalogDictionary.OC_PROPERTIES]
```

This code works as expected, however, `mypy` 1.5.1 doesn't like the second line and tells me `Value of type "PdfObject" is not indexable`. From what I've seen so far, `pypdf` uses more or less the same code in several places and `mypy` does not complain. 

Any idea why I could be seeing this?

**Comment by pubpub-zz** [comment, marked as answer]
the annotation of the get(`[ ]`) of a DictionaryObject such as trailer indicates the generic common type PdfObject, to prevent errors, you need to cast it to the adequate type):
`root = cast(DictionaryObject,reader.trailer[TrailerKeys.ROOT])`
you may neeed to add a global  `from pypdf.generic import *`

**Reply by rassie** [reply]
Thanks! I thought `PdfObject` was a descendant of `DictionaryObject`, but apparently it's not.
